File: cpp/src/Aves.cpp
#include "../include/Aves.h"
#include <iostream>

Aves::Aves(const std::string& identificacao, int idade) {
    this->identificacao = identificacao;
    this->idade = idade;
}

int Aves::totalFerteis() {
    return (idade > 1 && idade < 3) ? 1 : 0;
}

void Aves::manejar() {
    std::cout << "Manejo específico das aves." << std::endl;
    for (const auto& t : tratamentos) {
        std::cout << "- Data: " << t.getData() << ", Tipo: " << t.getTipo() 
                  << ", Produto: " << t.getProduto() << std::endl;
    }
    std::cout << std::endl;
}

File: python/aves.py
from animal import Animal

class Aves(Animal):
    def __init__(self, identificacao, idade):
        super().__init__()
        self.identificacao = identificacao
        self.idade = idade

    def total_ferteis(self):
        if 1 < self.idade < 3:
            return 1
        else:
            return 0

    def manejar(self):
        print("Manejo específico das aves.")
        for t in self.tratamentos:
            print(f"- Data: {t.get_data()}, Tipo: {t.get_tipo()}, Produto: {t.get_produto()}")
        print()